ch_pipeline/processing/client.py

```diff
@@ -229,6 +235,32 @@ def generate(revision, number, max_number, submit, fairshare, user_fairshare):
     revision.generate(max=number_to_submit, submit=submit)
 
 
+@metrics.command("list")
+@click.argument("revision", type=PREV)
+def metrics_list(revision):
+    """Show metrics about currently running jobs for
+    REVISION (given as (type:revision)."""
+
+    fs = base.slurm_fairshare("rpp-chime_cpu")
+    ls = revision.ls()
+    available = revision.available()
+    waiting, running = revision.queued()
+    success, failed = revision.completed()
+    # Direct copy from revision.pending method, put here
+    # to avoid duplicate calls
+    not_pending = set(ls) | set(waiting) | set(running)
+    pending = [job for job in available if job not in not_pending]
+
+    click.echo("Fairshare: {0}".format(fs[0]))
+    click.echo("User Fairshare: {0}".format(fs[1]))
+    click.echo("Available: {0}".format(len(available)))
+    click.echo("Pending: {0}".format(len(pending)))
+    click.echo("Waiting: {0}".format(len(waiting)))
+    click.echo("Running: {0}".format(len(running)))
+    click.echo("Successful: {0}".format(len(success) + len(ls)))
+    click.echo("Failed: {0}".format(len(failed)))
```

Comment:
Stylistically I would definitely advocate for f-strings in here. Shorter and cleaner IMO (and apparently faster too).

You might also want to consider if the output should be in some more structured format to making later parsing easier.

Reply:
Fixed in b31ece6. My thought process for output format is that this should be more about user readability, while any formatting/parsing logic should be done elsewhere